From 4505c22689e5865cb5f76830ffc01266317b3894 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 6 Oct 2026 07:08:54 +0200
Subject: [PATCH] avformat/dashdec: Check that the referenced representation
 has an id

Fixes: NULL pointer dereference
Fixes: pBjbWIWDyAJK
NULL pointer dereference Replicated through UnModified FFmpeg (ffprobe) with UBSAN
Regression since: e7696357de
Found during triage of the security report XmvwOQImSoeX
---
 libavformat/dashdec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/dashdec.c b/libavformat/dashdec.c
index ef33ff5ff397..0911bd73fad0 100644
--- a/libavformat/dashdec.c
+++ b/libavformat/dashdec.c
@@ -2296,7 +2296,7 @@ static int dash_read_header(AVFormatContext *s)
             if (!ref->nb_assoc_stream)
                 continue;
             const AVDictionaryEntry *id = av_dict_get(ref->assoc_stream[0]->metadata, "id", NULL, AV_DICT_MATCH_CASE);
-            if (!strcmp(rep->dependencyid, id->value))
+            if (id && !strcmp(rep->dependencyid, id->value))
                 break;
         }
         if (j >= c->n_videos || !av_strstart(rep->codecs, "lvc1", NULL) ||
